src/service/viewer-impl.js

```diff
@@ -802,6 +802,25 @@ export class Viewer {
   }
 
   /**
+   * Get the fragment from the url or the viewer.
+   * Strip leading '#' in the fragment
+   * @return {!Promise<string|undefined>}
+   */
+  getFragment() {
+    if (!this.isEmbedded_) {
+      let hash = this.win.location.hash;
+      if (hash.indexOf('#') == 0) {
```

Comment:
We can skip this check entirely. If there is a hash, it will always start with a `#`. If there's no hash, the `#substr` is a noop.